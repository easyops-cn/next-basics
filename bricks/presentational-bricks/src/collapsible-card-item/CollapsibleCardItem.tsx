import React, { CSSProperties } from "react";
import { MenuIcon } from "@next-core/brick-types";
import { GeneralIcon } from "@next-libs/basic-components";
import classNames from "classnames";

interface CollapsibleCardItemProps {
  cardTitle: string;
  cardDesc?: string;
  isActive?: boolean;
  icon?: MenuIcon;
  iconStyle?: Record<string, any>;
  customHeader?: boolean;
  contentStyle?: Record<string, any>;
  showSubscript?: boolean;
  subscriptIconConfig?: MenuIcon;
  hideOperate?: boolean;
  hoverable?: boolean;
  cardStyle?: CSSProperties;
  operatingAreaStyle?: React.CSSProperties;
}

export function CollapsibleCardItem(
  props: CollapsibleCardItemProps
): React.ReactElement {
  return (
    <div
      className={classNames("brickCollapsibleCardContainer", {
        ["hoverable"]: props.hoverable,
      })}
      style={props.cardStyle}
    >
      <div
        className={classNames(
          "headerContainer"
          // props.isActive ? "borderButtom" : null
        )}
        style={
          props.isActive
            ? {
                borderBottom:
                  "var(--border-width-base) solid var(--color-text-divider-line)",
              }
            : {}
        }
      >
        {props.showSubscript && (
          <div className="subscriptContainer">
            <GeneralIcon icon={props.subscriptIconConfig}></GeneralIcon>
          </div>
        )}
        <div className="collapsibleCardItemInfo">
          {props.customHeader ? (
            <slot id="headerSlot" name="header" />
          ) : (
            <>
              {props.icon && (
                <i style={props.iconStyle} className="collapsibleCardItemIcon">
                  <GeneralIcon icon={props.icon} />
                </i>
              )}
              <div className="collapsibleTitle">
                {props.cardTitle}
                {props.cardDesc && <>（{props.cardDesc}）</>}
              </div>
            </>
          )}
        </div>
        {!props.hideOperate && (
          <div className="operatingArea" style={props.operatingAreaStyle}>
            <slot id="operateSlot" name="operate" />
          </div>
        )}
      </div>
      {props.isActive && (
        <div className="contentContainer" style={props.contentStyle}>
          <slot id="contentSlot" name="content" />
        </div>
      )}
    </div>
  );
}
