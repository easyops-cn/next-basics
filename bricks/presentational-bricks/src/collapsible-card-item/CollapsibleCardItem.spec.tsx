import React from "react";
import { shallow } from "enzyme";
import { CollapsibleCardItem } from "./CollapsibleCardItem";

describe("CollapsibleCardItem", () => {
  it("should work", () => {
    const wrapper = shallow(
      <CollapsibleCardItem
        cardTitle="gitlab"
        cardDesc="Easyops GIT"
        icon={{
          lib: "antd",
          type: "gitlab",
        }}
        hoverable={false}
        cardStyle={{ border: "none" }}
        isActive={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({
      isActive: false,
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
