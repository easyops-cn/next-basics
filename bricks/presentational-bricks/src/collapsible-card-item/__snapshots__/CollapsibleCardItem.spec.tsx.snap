// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleCardItem should work 1`] = `
<div
  className="brickCollapsibleCardContainer"
  style={
    Object {
      "border": "none",
    }
  }
>
  <div
    className="headerContainer"
    style={
      Object {
        "borderBottom": "var(--border-width-base) solid var(--color-text-divider-line)",
      }
    }
  >
    <div
      className="collapsibleCardItemInfo"
    >
      <i
        className="collapsibleCardItemIcon"
      >
        <GeneralIcon
          icon={
            Object {
              "lib": "antd",
              "type": "gitlab",
            }
          }
        />
      </i>
      <div
        className="collapsibleTitle"
      >
        gitlab
        （
        Easyops GIT
        ）
      </div>
    </div>
    <div
      className="operatingArea"
    >
      <slot
        id="operateSlot"
        name="operate"
      />
    </div>
  </div>
  <div
    className="contentContainer"
  >
    <slot
      id="contentSlot"
      name="content"
    />
  </div>
</div>
`;

exports[`CollapsibleCardItem should work 2`] = `
<div
  className="brickCollapsibleCardContainer"
  style={
    Object {
      "border": "none",
    }
  }
>
  <div
    className="headerContainer"
    style={Object {}}
  >
    <div
      className="collapsibleCardItemInfo"
    >
      <i
        className="collapsibleCardItemIcon"
      >
        <GeneralIcon
          icon={
            Object {
              "lib": "antd",
              "type": "gitlab",
            }
          }
        />
      </i>
      <div
        className="collapsibleTitle"
      >
        gitlab
        （
        Easyops GIT
        ）
      </div>
    </div>
    <div
      className="operatingArea"
    >
      <slot
        id="operateSlot"
        name="operate"
      />
    </div>
  </div>
</div>
`;
